cluster-autoscaler/cloudprovider/aws/README.md

```diff
@@ -71,153 +92,158 @@ env:
   value: YOUR_AWS_REGION
 ```
 
-## Deployment Specification
-Auto-Discovery Setup is always preferred option to avoid multiple, potentially different configuration for min/max values. If you want to adjust minimum and maximum size of the group, please adjust size on ASG directly, CA will fetch latest change when talking to ASG.
+## Auto-Discovery Setup
 
-If you use one or multiple ASG setup, the min/max configuration change in CA will not make the corresponding change to ASG. Please make sure CA min/max values are within the boundary of ASG minSize and maxSize.
+Auto-Discovery Setup is the preferred method to configure Cluster Autoscaler.
 
-### One ASG Setup (min: 1, max: 10, ASG Name: k8s-worker-asg-1)
-```
-kubectl apply -f examples/cluster-autoscaler-one-asg.yaml
-```
+To enable this, provide the `--node-group-auto-discovery` flag as an argument
+whose value is a list of tag keys that should be looked for. For example,
+`--node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/<cluster-name>`
+will find the ASGs where those tag keys _exist_. It does not matter what value
+the tags have.
 
-### Multiple ASG Setup
-```
-kubectl apply -f examples/cluster-autoscaler-multi-asg.yaml
-```
-
-### Master Node Setup
+Example deployment:
 
-To run a CA pod in master node - CA deployment should tolerate the master `taint` and `nodeSelector` should be used to schedule the pods in master node.
-Please replace `{{ node_asg_min }}`, `{{ node_asg_max }}` and `{{ name }}` with your ASG setting in the yaml file.
 ```
-kubectl apply -f examples/cluster-autoscaler-run-on-master.yaml
+kubectl apply -f examples/cluster-autoscaler-autodiscover.yaml
 ```
 
-### Auto-Discovery Setup
+Cluster Autoscaler will respect the minimum and maximum values of each Auto
+Scaling Group. It will will only adjust the desired value.
 
-To run a cluster-autoscaler which auto-discovers ASGs with nodes use the `--node-group-auto-discovery` flag. For example, `--node-group-auto-discovery=asg:tag=k8s.io/cluster-autoscaler/enabled,k8s.io/cluster-autoscaler/<YOUR CLUSTER NAME>` will find the ASGs where those tag keys
-_exist_. It does not matter what value the tags have.
+Each Auto Scaling Group should be comprised of instance types that provide
```

Comment:
Even mixed instance policy is supported. This is not something we recommend users to use. It never guarantee all the simulation is accurate because if instances ASG bring up are unknown to CA. I don't recommend we talk about this here. It would be better to move to MixedInstancePolicy section. The best practice is still one ASG -> one Launch Template -> One Instance type.

Reply:
I agree that using a single ASG with a single instance type is the simplest case for most customers, and we can come up with verbiage to emphasize that. However, we do recommend using Mixed Instance policies under several circumstances: (1) customers using Spot Instances, and (2) very large customers who benefit from instance diversity to address node shortages in certain regions/AZs. 

I disagree that a single ASG is "best practice" however, since we even support multiple ASGs with Managed Node Groups out of the box and encourage customers to use it where it makes sense for them.

Reply:
Mixed Instance policy is not mature enough as a solution and user needs to take the risk and have to know all prerequisites to use it. It's just one presentation of spot, there're group of users using spot separately without MIP. I may not explain it clearly, my point is single instance per ASG is best practice, not single ASG. You can definitely has more ASGs/node groups.